src/main/java/hudson/remoting/forward/CopyThread.java

```diff
@@ -20,6 +20,9 @@ public CopyThread(String threadName, InputStream in, OutputStream out) {
         super(threadName);
         this.in = in;
         this.out = out;
+        setUncaughtExceptionHandler((t, e) -> {
+            LOGGER.log(Level.WARNING, "Uncaught exception in CopyThread " + t, e);
+        });
```

Comment:
If we're restarting the thread, this should probably also used a buffered input and output stream, no?  Otherwise data will be lost or mangled.